src/eos-metrics-instrumentation.c

```diff
@@ -75,6 +86,94 @@ static gint64 start_time;
 
 static EinsPersistentTally *persistent_tally;
 
+static gboolean
+get_os_version (gchar **name_out,
+                gchar **version_out)
+{
+    g_autoptr(GError) error = NULL;
+    gboolean ret = FALSE;
+
+    g_autoptr(GFile) os_release_file = g_file_new_for_path (OS_RELEASE_FILE);
+    g_autoptr(GFileInputStream) file_stream = g_file_read (os_release_file, NULL, &error);
+    if (error)
+      goto out;
+
+    g_autoptr(GDataInputStream) data_stream = g_data_input_stream_new (G_INPUT_STREAM (file_stream));
+
+    g_autofree gchar *name = NULL;
+    g_autofree gchar *version = NULL;
+    gchar *line;
+    while ((!name || !version) &&
+           (line = g_data_input_stream_read_line (data_stream, NULL, NULL, &error)))
+      {
+        if (g_str_has_prefix (line, "NAME="))
+          name = line;
+        else if (g_str_has_prefix (line, "VERSION="))
+          version = line;
+        else
+          g_free (line);
+      }
+
+    if (error)
+      goto out;
+    if (!name || !version)
+      {
+        g_warning ("Could not find NAME or VERSION keys in " OS_RELEASE_FILE ".");
```

Comment:
nit: This error message can be interpreted to mean that neither the `NAME` nor the `VERSION` keys are present; when really the issue is that at least one of them couldn't be found.